ap.md

```diff
@@ -0,0 +1,236 @@
+    var req = new XMLHttpRequest();
+
+    // Open a GET request to '/leds/:index'
+    req.open('GET', '/leds/' + ledIndex);
+
+    // Once the request gets a successful response, update that statusNode with the status of the LED.
+    req.onload = function(e) {
+      if (req.readyState == 4 && req.status == 200) {
+        if (req.status == 200) {
+          var response = JSON.parse(req.responseText);
+          statusNode.textContent = response.on ? 'ON' : 'OFF';
+        } else { 
+          console.log('Error'); 
+        }
+      }
+    }
+    req.send(); // send our request to the server
+  }
+</script>
+{% endhighlight %}
+
+
+While there are some comments in the previous code snippet, the summary is when a button with the class "led-button" is clicked, we grab the info from its "data-led" attribute and make a request to our Tessel server with that info in the url. After we get a successful response from the server, we update the "Status" field next to the button with the state of the corresponding LED. 
+
+Now let's check out `ap.js` again to finish up the project. First, we're going to make a few changes to our server setup:
+
+{% highlight javascript %}
+// These two dependencies remain the same
+var tessel = require('tessel');
+var http = require('http');
+
+// Require two other core Node.js modules
+var fs = require('fs');
+var url = require('url');
+
+var server = http.createServer(function (request, response) {
+  // here is the biggest change
+  var urlParts = url.parse(request.url, true); // break up the url into easier-to-use parts
+  var ledRegex = /leds/; // a Regex to catch requests to toggle LEDs
+
+  if (urlParts.pathname.match(ledRegex)) {
+    // if there is a request containing 'leds' call a function, toggleLED
+    toggleLED(urlParts.pathname, request, response);
+  } else {
+    // all other request will call a function, showIndex
+    showIndex(urlParts.pathname, request, response);
+  }
+});
+
+// Stays the same
+server.listen(8080);
+
+// Stays the same
+console.log('Server running at http://192.168.1.101:8080/');
+{% endhighlight %}
+
+Inside our `createServer` callback function, we've replaced the old plain text response with a basic router that catches requests to '/leds/:index' and defaults all other requests to a separate function. Let's add those functions below the `console.log`:
+
+{% highlight javascript %}
+// respond to the request with our index.html page
+function showIndex (url, request, response) {
+  response.writeHead(200, {"Content-Type": "text/html"});
+  // use fs to read in index.html
+  fs.readFile(__dirname + '/index.html', function (err, content) {
+    if (err) {
+      throw err;
+    }
+
+    // serves the content of index.html read in by fs
+    response.end(content);
+  });
+}
+
+// toggle the led specified in the url and respond with its state
+function toggleLED (url, request, response) {
+  var indexRegex = /(\d)$/; // Regex to find the number at the end of the url
```

Comment:
"Find the number at the end of the URL with a regular expression" for parallel construction with other comments